submissions/krylenger/friends/app.js

```diff
@@ -0,0 +1,259 @@
+};
+
+const emptyCardsContainer = () => {
+  cardsContainer.innerHTML = '';
+}
+
+const renderUserCards = (usersData) => {
+  emptyCardsContainer();
+  const fragment = document.createDocumentFragment();
+  usersData.forEach((user) => {
+    const userCard = createCard(user);
+    fragment.appendChild(userCard);
+  });
+  cardsContainer.appendChild(fragment);
+};
+
+const sortByParam = (usersData, sortingCriteria, direction) => {
+  return usersData.sort((firstElem, secondElem) => {
+    let a, b;
+    
+    switch (sortingCriteria) {
+      case 'name':
+        a = firstElem.name.first;
+        b = secondElem.name.first;
+        break;
+      case 'age':
+        a = firstElem.dob.age;
+        b = secondElem.dob.age;
+        break;
+      case 'registrationDate':
+        a = firstElem.registered.date;
+        b = secondElem.registered.date;
+        break;
+    }
+
+     if (a < b) {
+       return direction === 'ascending' ? -1 : 1;
+     }
+     if (a > b) {
+      return direction === 'ascending' ? 1 : -1;
+     }
+    return 0;
+  });
+}
+
+const doSearch = (usersData, searchedValue) => {
+  return usersData.filter((user) => {
+    const fullName = createFullName(user.name.first, user.name.last);
+    return fullName.toLocaleLowerCase().includes(searchedValue);
+  })
+}
+
+const doFilterByGender = (usersData, gender) => usersData.filter(user => user.gender === gender);
+
+const doFilterByCountry = (usersData, country) => usersData.filter(user => user.location.country === country);
+
+const doFilterByAge = (usersData, age) => usersData.filter(user => user.dob.age <= age);
+
+
+const filterCards = (usersData) => {
+  let filteredUsersData = usersData;
+
+  if (filtersState.search) {
+    filteredUsersData = doSearch(filteredUsersData, filtersState.search);
+  }
+  if (filtersState.gender) {
+    filteredUsersData = doFilterByGender(filteredUsersData, filtersState.gender);
+  }
+  if (filtersState.country) {
+    filteredUsersData = doFilterByCountry(filteredUsersData, filtersState.country);
+  }
+  if (filtersState.age) {
+    filteredUsersData = doFilterByAge(filteredUsersData, filtersState.age);
+  }
+  if (filtersState.sortName === 'name_AZ') {
+    filteredUsersData = sortByParam(filteredUsersData, 'name', 'ascending');
+  }
+  if (filtersState.sortName === 'name_ZA') {
+    filteredUsersData = sortByParam(filteredUsersData, 'name', 'descending');
+  }
+  if (filtersState.sortAge === 'age_19') {
+    filteredUsersData = sortByParam(filteredUsersData, 'age', 'ascending');
+  }
+  if (filtersState.sortAge === 'age_91') {
+    filteredUsersData = sortByParam(filteredUsersData, 'age', 'descending');
+  }
+  if (filtersState.sortRegistrationDate === 'registrationDate_NEW') {
+    filteredUsersData = sortByParam(filteredUsersData, 'registrationDate', 'ascending');
+  }
+  if (filtersState.sortRegistrationDate === 'registrationDate_OLD') {
+    filteredUsersData = sortByParam(filteredUsersData, 'registrationDate', 'descending');
+  } 
```

Comment:
Switch/Case will be better then than many if's

Reply:
I've tried to implement switch/case here but didn't succeed. How to deal with 'break'? 
Because in my app I can apply several filters at the same time but inside switch/case we have to use 'break' to exit the statement.